Framework/DataHandling/src/LoadSampleEnvironment.cpp

```diff
@@ -313,32 +317,68 @@ boost::shared_ptr<MeshObject> LoadSampleEnvironment::translate(
 }
 
 /**
- * Rotates the environment by a provided matrix
+ * Rotates the environment by a generated matrix
  * @param environmentMesh The environment to rotate
  * @returns a shared pointer to the newly rotated environment
  */
 boost::shared_ptr<MeshObject>
 LoadSampleEnvironment::rotate(boost::shared_ptr<MeshObject> environmentMesh) {
-  const std::vector<double> rotationMatrix = getProperty("RotationMatrix");
-  double valueList[] = {1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0, 0.0, 0.0};
-  std::vector<double> checkVector1 =
-      std::vector<double>(std::begin(valueList), std::end(valueList));
-  if (rotationMatrix != checkVector1) {
-    if (rotationMatrix.size() != 9) {
-
-      throw std::invalid_argument(
-          "Invalid Rotation Matrix, must have exactly 9 values, not: " +
-          std::to_string(rotationMatrix.size()));
-    }
-    Matrix<double> rotation = Matrix<double>(rotationMatrix);
-    double determinant = rotation.determinant();
-    if (!(std::abs(determinant) == 1.0)) {
-      throw std::invalid_argument("Invalid Rotation Matrix");
-    }
-    environmentMesh->rotate(rotation);
-  }
+  const std::vector<double> rotationMatrix = generateMatrix();
+  environmentMesh->rotate(rotationMatrix);
   return environmentMesh;
 }
 
+/**
+ * Generates a rotation Matrix applying the x rotation then y rotation, then z
+ * rotation
+ * @returns a matrix of doubles to use as the rotation matrix
+ */
+Matrix<double> LoadSampleEnvironment::generateMatrix() {
+  Kernel::Matrix<double> xMatrix = generateXRotation();
+  Kernel::Matrix<double> yMatrix = generateYRotation();
+  Kernel::Matrix<double> zMatrix = generateZRotation();
+
+  return zMatrix * yMatrix * xMatrix;
+}
+
+/**
+ * Generates the x component of the rotation matrix
+ * using the xDegrees Property.
+ * @returns a matrix of doubles to use as the x axis rotation matrix
+ */
+Matrix<double> LoadSampleEnvironment::generateXRotation() {
+  const double xRotation = getProperty("xDegrees");
```

Comment:
This can compute the radians, and use them below, to avoid computing them twice in `sin` and `cos`. Same for Y and Z rotations